File: client-clojure/src/sirqul_io_t_platform/specs/profile_short_response.clj
(ns sirqul-io-t-platform.specs.profile-short-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            )
  (:import (java.io File)))


(def profile-short-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :accountId) int?
   (ds/opt :locationDisplay) string?
   (ds/opt :display) string?
   (ds/opt :username) string?
   (ds/opt :profileImage) string?
   (ds/opt :profileAvatar) asset-short-response-spec
   (ds/opt :returning) string?
   })

(def profile-short-response-spec
  (ds/spec
    {:name ::profile-short-response
     :spec profile-short-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ProfileShortResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ProfileShortResponse._

case class ProfileShortResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
accountId: Option[Long],
locationDisplay: Option[String],
display: Option[String],
username: Option[String],
profileImage: Option[String],
profileAvatar: Option[AssetShortResponse],
returning: Option[String])

object ProfileShortResponse {
  import DateTimeCodecs._

  implicit val ProfileShortResponseCodecJson: CodecJson[ProfileShortResponse] = CodecJson.derive[ProfileShortResponse]
  implicit val ProfileShortResponseDecoder: EntityDecoder[ProfileShortResponse] = jsonOf[ProfileShortResponse]
  implicit val ProfileShortResponseEncoder: EntityEncoder[ProfileShortResponse] = jsonEncoderOf[ProfileShortResponse]
}
